Add Python worker versioning guide (#2194)
temporalio/documentation: docs-src/python/how-to-use-worker-versioning-in-python.md

---
id: how-to-use-worker-versioning-in-python
title: How to use Worker Versioning in Python
sidebar_label: Worker Versioning
description: Version your Python Workers by using build ID-based versioning
tags:
  - python
  - how-to
---

> To use [Worker Versioning](/workers#worker-versioning) in Python, you need to do the following:

Reviewer: May need a bit more obvious link to the concepts section as it can almost be a prerequisite before getting into Python specifics

1. Determine and assign a Build ID to your built Worker code, and opt in to versioning.

> 2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID.

Reviewer: ```suggestion
2. Tell the Task Queue your Worker is listening on that Build ID, and whether it's compatible with an existing Build ID.
```

```python
# ...
await client.update_worker_build_id_compatibility(
    "your_task_queue_name", BuildIdOpAddNewCompatible("deadbeef", "some-existing-build-id")
)
```

This code adds `deadbeef` to the existing compatible set containing `some-existing-build-id` and marks it as the new default Build ID for that set.

You can also promote an existing Build ID in a set to be the default for that set:

```python
# ...
await client.update_worker_build_id_compatibility(
    "your_task_queue_name", BuildIdOpPromoteBuildIdWithinSet("deadbeef")
)
```

You can also promote an entire set to become the default set for the queue. New Workflows will start using that set's default build.

```python
# ...
await client.update_worker_build_id_compatibility(
    "your_task_queue_name", BuildIdOpPromoteSetByBuildId("deadbeef")
)
```

### Specify versions for Commands

By default, Activities, Child Workflows, and Continue-as-New use the same compatible version set as
the Workflow that invoked them if they're also using the same Task Queue.

If you want to override this behavior, you can specify your intent via the `versioning_intent`
argument available on the methods you use to invoke these Commands.

> <!-- For more information refer to the [conceptual documentation](/concepts/what-is-worker-versioning). -->

Reviewer: Uncomment this, I think it's important context
